src/engine/world/room.cpp

```diff
@@ -584,12 +585,39 @@ void Room::collectShaderLights(size_t depth)
   lightsBuffer->setData(bufferLights, gl::api::BufferUsage::StaticDraw);
 }
 
-void Room::createParticleMesh(const std::string& label,
-                              const glm::vec3& min,
-                              const glm::vec3& max,
-                              render::scene::MaterialManager& materialManager)
+void Room::regenerateDust(const std::shared_ptr<engine::Presenter>& presenter,
+                          const gslu::nn_shared<render::scene::Material>& dustMaterial,
+                          bool isDustEnabled,
+                          float dustDensityDivisor)
+{
+  if(!isDustEnabled)
+  {
+    dust = nullptr;
+    return;
+  }
+
+  if(const auto it = dustCache.find(dustDensityDivisor); it != dustCache.end())
+  {
+    dust = it->second;
+    return;
+  }
+  if(presenter != nullptr)
+    presenter->drawLoadingScreen(_("Generating Dust Particles..."));
+  auto dustNode = createParticleMesh(node->getName() + "/dust", verticesBBoxMin, 
```

Comment:
General guideline: if the argument list is split across multiple lines, there's only one argument on each line.